packages/utils/package.json

```diff
@@ -0,0 +1,42 @@
+{
+  "name": "@unocss/utils",
+  "version": "0.55.7",
+  "description": "Utilities for UnoCSS",
+  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
+  "license": "MIT",
+  "funding": "https://github.com/sponsors/antfu",
+  "homepage": "https://github.com/unocss/unocss/tree/main/packages/utils#readme",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/unocss/unocss",
+    "directory": "packages/utils"
+  },
+  "bugs": {
+    "url": "https://github.com/unocss/unocss/issues"
+  },
+  "keywords": [],
+  "sideEffects": false,
+  "exports": {
+    ".": {
+      "types": "./dist/index.d.ts",
+      "import": "./dist/index.mjs",
+      "require": "./dist/index.cjs"
+    }
+  },
+  "main": "dist/index.cjs",
+  "module": "dist/index.mjs",
+  "types": "dist/index.d.ts",
+  "files": [
+    "dist"
+  ],
+  "engines": {
+    "node": ">=14"
+  },
+  "scripts": {
+    "build": "unbuild",
+    "stub": "unbuild --stub"
+  },
+  "dependencies": {
+    "@unocss/core": "workspace:^"
```

Comment:
As a utils package, we shouldn't have other dependencies. 
Instead, we should migrate the functions in `@unocss/core` to `@unocss/utils`, and import `@unocss/utils` dependencies in `@unocss/core`.

Reply:
Agree

Reply:
Well, I am thinking the other way around. If some utils are used in core, then let them stay in core. `@unocss/utils` should be pictured as "utils for creating rules/presets", so it can have tools like color parsing etc. While the `core` is pretty much a CSS-agnostic engine that just caring about extracting and conacting strings.

I'd prefer `@unocss/utils` to deps on `@unocss/core` and keep `@unocss/core` deps-free.

If the name is a bit confusing, maybe we could have `@unocss/rule-utils`.

Reply:
`@unocss/rule-utils` Seems good to me. What do you think? @zyyv

Reply:
Now rename to `@unocss/rule-utils`